File: Solutions/1380. Lucky Numbers in a Matrix/1380. Lucky Numbers in a Matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] mat) {
        int m=mat.length,n=mat[0].length;
        int maxMin=Integer.MIN_VALUE;
        for(int i=0;i<m;++i){
            int Min=Integer.MAX_VALUE;
            for(int j=0;j<n;++j)
                Min=Math.min(Min,mat[i][j]);
            maxMin=Math.max(maxMin,Min);
        }
        int minMax=Integer.MAX_VALUE;
        for(int j=0;j<n;++j){
            int Max=Integer.MIN_VALUE;
            for(int i=0;i<m;++i)
                Max=Math.max(Max,mat[i][j]);
            minMax=Math.min(minMax,Max);
        }
        var res=new ArrayList<Integer>();
        if(minMax!=maxMin)
            return res;
        res.add(minMax);
        return res;
    }
}

File: Solutions/1380. Lucky Numbers in a Matrix/1380. Lucky Numbers in a Matrix.cpp
class Solution {
public:
    vector<int> luckyNumbers (vector<vector<int>>& mat) {
        int m=mat.size(),n=mat[0].size();
        int maxMin=INT_MIN;
        for(int i=0;i<m;++i){
            int Min=INT_MAX;
            for(int j=0;j<n;++j)
                Min=min(Min,mat[i][j]);
            maxMin=max(maxMin,Min);
        }
        int minMax=INT_MAX;
        for(int j=0;j<n;++j){
            int Max=INT_MIN;
            for(int i=0;i<m;++i)
                Max=max(Max,mat[i][j]);
            minMax=min(minMax,Max);
        }
        if(minMax!=maxMin)
            return vector<int> {};
        return vector<int> {maxMin};
    }
};